### PR TITLE
Distinguish known empty user profiles in consent filtering

### LINKED ISSUE #5098: Fix consent attribute filtering for known empty user profiles
## Summary

The consent attribute-profile filter does not distinguish between unavailable profile data and a known empty user profile.

`buildAttributePurposePrompt` currently filters an attribute only when `len(userAttributeSet) > 0`. In Go, a nil map and an empty map both have length zero. As a result, when the identity provider returns a known empty attribute set, the consent flow disables profile filtering and can prompt for requested attributes that are not present for the user.

## Required change

Define and implement the intended profile-filter contract in the default consent service:

- If profile data is available and contains no applicable attributes, do not create an attribute consent prompt.
- Preserve the behavior for profile formats that do not expose individual attribute names, if that behavior remains required.
- Distinguish an unavailable or opaque profile from a known empty profile.
- Add focused tests for nil, empty, opaque JWT/JWE, and populated attribute responses.

## Affected area

- `backend/internal/authn/consent/service.go`
- Consent prompt construction in `buildUserAttributeSet` and `buildAttributePurposePrompt`

## Rationale

The default consent service should not request consent for attributes that are known to be absent from the user profile.

## Acceptance criteria

- A known empty `providers.AttributesResponse.Attributes` set produces no attribute consent prompt.
- A populated profile only produces prompts for attributes present in the profile.
- The behavior for opaque JWT/JWE attribute responses is explicit and covered by tests.
- The implementation does not include the PR-specific `_jwt` customization proposed in pull request #5085.

## Backlinks

- Pull request: https://github.com/thunder-id/thunderid/pull/5085
- Review comment: https://github.com/thunder-id/thunderid/pull/5085#discussion_r3811737342
- Requested by: @ThaminduDilshan

### PR DESCRIPTION
**Pull request by sandaruwijewardhana**
### Purpose
An empty profile map ({}) was previously treated the same as a missing profile (nil). This disabled attribute filtering entirely which causing users to be prompted for attributes they didn't have.
The internal _jwt pseudo claim was treated as a regular attribute name which causing all requested attributes to be filtered out and skipping consent collection completely.

### Approach
Updated buildUserAttributeSet and buildAttributePurposePrompt to distinguish between an unknown profile and a known empty profile
The buildUserAttributeSet now returns nil only when the profile is missing or opaque (providers.RawJWTAttributeKey). For a known profile (even if empty) it returns a non-nil set.
The buildAttributePurposePrompt now checks userAttributeSet != nil instead of len(userAttributeSet) > 0 that ensuring empty profiles are properly filtered.
The buildUserAttributeSet has a single production caller and its result is consumed in one place and both within service.go. So the contract change does not propagate beyond that file.

### Related Issues
Fixes #5098

### Related PRs
N/A

### Checklist
- [x] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [x] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [x] Tests provided. (Add links if there are any)
    - [x] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

* **Bug Fixes**
  * Consent prompts now show only attributes available in a user’s profile.
  * Empty profiles no longer display profile-specific consent elements.
  * Profile filtering is skipped when profile details are unavailable or provided only as an opaque token.

* **Documentation**
  * Added guidance on attribute filtering and consent prompt behavior.

* **Tests**
  * Added coverage for empty, populated, unavailable, and opaque profile scenarios.

### CODE REVIEW

File: backend/internal/authn/consent/service.go [hunk 1]
```diff
@@ -586,7 +594,7 @@ func buildAttributePurposePrompt(purpose consent.ConsentPurpose,
 		}
 
 		// Skip elements not present in the user profile
-		if len(userAttributeSet) > 0 && !userAttributeSet[elem.Name] {
+		if userAttributeSet != nil && !userAttributeSet[elem.Name] {
```
[the remark this answers is not in the document]
Comment (sandaruwijewardhana) [reply backend/internal/authn/consent/service.go:597]: Added a paragraph to docs/content/guides/consent.mdx under "Attribute Consent in Login Flows" clarifying three profile cases ->Missing attributes - Omitted from prompt,  Empty profile - Prompts for nothing, Opaque token profile - Prompts for everything.

**Comment by sandaruwijewardhana** [comment]
@ThaminduDilshan the workflows on this PR haven't been triggered and only DCO has run . It looks like they waiting on maintainer approval for a first time contributor. Could you approve the CI run?

Also docs comment is addressed and added a paragraph to docs/content/guides/consent.mdx covering known empty, populated and opaque profiles.
